jdaviz/configs/rampviz/plugins/viewers.py

```diff
@@ -0,0 +1,140 @@
+                ]
+
+    default_class = NDDataArray
+    _default_profile_subset_type = 'temporal'
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('default_tool_priority', ['jdaviz:selectslice'])
+        super().__init__(*args, **kwargs)
+
+    def _initialize_x_axis(self):
+        if len(self.state.x_att_helper.choices):
+            self.state.x_att = self.state.x_att_helper.choices[-1]
+            self.set_plot_axes()
+            self.reset_limits()
+
+    def reset_limits(self):
+        super().reset_limits()
+
+        # override to reset to the global y limits including marks:
+        global_y_min = float(self.state.y_min)
+        global_y_max = float(self.state.y_max)
+        for mark in self.figure.marks:
+            if len(mark.y) and mark.visible:
+                global_y_min = min(global_y_min, np.nanmin(mark.y))
+                global_y_max = max(global_y_max, np.nanmax(mark.y))
+
+        if global_y_min != self.state.y_min or global_y_max != self.state.y_max:
+            self.set_limits(
+                y_min=global_y_min * 0.9,
+                y_max=global_y_max * 1.1
+            )
+
+    def set_plot_axes(self):
+
+        with self.figure.hold_sync():
+            self.figure.axes[0].label = "Group"
+            self.figure.axes[1].label = self.state.y_display_unit
+
+            # Make it so axis labels are not covering tick numbers.
+            self.figure.fig_margin["left"] = 95
+            self.figure.fig_margin["bottom"] = 60
+            self.figure.send_state('fig_margin')  # Force update
+            self.figure.axes[0].label_offset = "40"
+            self.figure.axes[1].label_offset = "-70"
+            # NOTE: with tick_style changed below, the default responsive ticks in bqplot result
+            # in overlapping tick labels.  For now we'll hardcode at 8, but this could be removed
+            # (default to None) if/when bqplot auto ticks react to styling options.
+            self.figure.axes[1].num_ticks = 8
+
+            # Set Y-axis to scientific notation
+            self.figure.axes[1].tick_format = '0.1e'
+
+            for i in (0, 1):
+                self.figure.axes[i].tick_style = {'font-size': 15, 'font-weight': 600}
+
+
+@viewer_registry("rampviz-image-viewer", label="Image 2D (Rampviz)")
+class RampvizImageView(JdavizViewerMixin, WithSliceSelection, BqplotImageView):
+    # categories: zoom resets, (zoom, pan), subset, select tools, shortcuts
+    # NOTE: zoom and pan are merged here for space consideration and to avoid
+    # overflow to second row when opening the tray
+    tools_nested = [
+                    ['jdaviz:homezoom', 'jdaviz:prevzoom'],
+                    ['jdaviz:pixelboxzoommatch', 'jdaviz:boxzoom'],
+                    ['jdaviz:pixelpanzoommatch', 'jdaviz:panzoom'],
+                    ['bqplot:truecircle', 'bqplot:rectangle', 'bqplot:ellipse',
+                     'bqplot:circannulus'],
+                    ['jdaviz:rampperpixel'],
```

Comment:
somewhat unlelated, but I think we should revisit putting the per-pixel tool in the same group as the other subset tools (in cubeviz, and then also here).

Reply:
Personally I like them separate – should we make a ticket?